clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1025,23 +1025,6 @@ class KernelObjVisitor {
     VisitRecordFields(Owner, Handlers...);
   }
 
-  // FIXME: Can this be refactored/handled some other way?
-  template <typename ParentTy, typename... HandlerTys>
-  void visitStreamRecord(const CXXRecordDecl *Owner, ParentTy &Parent,
-                         CXXRecordDecl *Wrapper, QualType RecordTy,
-                         HandlerTys &... Handlers) {
-    (void)std::initializer_list<int>{
-        (Handlers.enterStream(Owner, Parent, RecordTy), 0)...};
```

Comment:
I think I know why __finalize call disappeared. It was previously generated by `leaveStream` function. Now it is not called, so the call is not generated.
I think what we need to do is:
- Remove enterStream/leaveStream functions as they are dead code now
- handleSpecialType should generate finalize statement if the class has `__finalize` method.